subsys/bluetooth/host/conn.c

```diff
@@ -1505,7 +1517,8 @@ void bt_conn_unref(struct bt_conn *conn)
 	 * to claim connection object as only the first claim will be served.
 	 */
 	if (deallocated) {
-		notify_recycled_conn_slot();
+		k_sem_give(&pending_recycled_events);
+		k_work_submit(&recycled_work);
```

Comment:
What happens if another connection gets deallocated before `recycled_work_handler` is run? 

`k_work_submit` would fail and the recycled callback would only be called once, even if multiple connection objects are deallocated, which doesn't seem to match the description of the callback.

Reply:
I was wondering something similar: maybe we implement a loop in the handler, until the semaphore count gets back to zero?

Reply:
The handler resubmits itself, so this doesn't seem like a concern. The handler could have a loop, but I think resubmitting is better since then we give other sys wq users to have their work items run in between.